tests/chainer_tests/functions_tests/math_tests/test_det.py

```diff
@@ -0,0 +1,217 @@
+            idt = numpy.array(idt).astype(numpy.float32)
+            chk = numpy.ones(self.x.shape[0]).astype(numpy.float32)
+        else:
+            idt = numpy.identity(self.x.shape[1]).astype(numpy.float32)
+            chk = numpy.ones(1).astype(numpy.float32)
+        if gpu:
+            chk = cuda.to_gpu(chk)
+            idt = cuda.to_gpu(idt)
+        idtv = chainer.Variable(idt)
+        idtd = self.det(idtv)
+        gradient_check.assert_allclose(idtd.data, chk, rtol=1e-4, atol=1e-4)
+
+    @attr.gpu
+    def test_det_identity_gpu(self):
+        self.det_identity(gpu=True)
+
+    def test_det_identity_cpu(self):
+        self.det_identity(gpu=False)
+
+    def det_product(self, gpu=False):
+        if gpu:
+            cx = cuda.to_gpu(self.x.copy())
+            cy = cuda.to_gpu(self.y.copy())
+        else:
+            cx = self.x.copy()
+            cy = self.y.copy()
+        vx = chainer.Variable(cx)
+        vy = chainer.Variable(cy)
+        dxy1 = self.det(self.matmul(vx, vy))
+        dxy2 = self.det(vx) * self.det(vy)
+        gradient_check.assert_allclose(dxy1.data, dxy2.data, rtol=1e-4,
+                                       atol=1e-4)
+
+    @condition.retry(3)
+    def test_det_product_cpu(self):
+        self.det_product(gpu=False)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_det_product_gpu(self):
+        self.det_product(gpu=True)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_batch_backward_gpu(self):
+        x_data = cuda.to_gpu(self.x.copy())
+        y_grad = cuda.to_gpu(self.gy.copy())
+        gradient_check.check_backward(self.det, x_data, y_grad)
+
+    @condition.retry(3)
+    def test_batch_backward_cpu(self):
+        x_data, y_grad = self.x.copy(), self.gy.copy()
+        gradient_check.check_backward(self.det, x_data, y_grad)
+
+    def test_expect_scalar_cpu(self):
+        x = numpy.random.uniform(.5, 1, (2, 2)).astype(numpy.float32)
+        x = chainer.Variable(x)
+        y = F.det(x)
+        self.assertEqual(y.data.ndim, 1)
+
+    @attr.gpu
+    def test_expect_scalar_gpu(self):
+        x = cuda.cupy.random.uniform(.5, 1, (2, 2)).astype(numpy.float32)
+        x = chainer.Variable(x)
+        y = F.det(x)
+        self.assertEqual(y.data.ndim, 1)
+
+    def test_zero_det_cpu(self):
+        x_data, y_grad = self.x.copy(), self.gy.copy()
+        if x_data.ndim == 3:
+            x_data[0, :, :] = 0.0
+        else:
+            x_data[:, :] = 0.0
+        with self.assertRaises(numpy.linalg.LinAlgError):
+            gradient_check.check_backward(self.det, x_data, y_grad)
+
+    @attr.gpu
+    def test_zero_det_gpu(self):
+        x_data = cuda.to_gpu(self.x.copy())
+        y_grad = cuda.to_gpu(self.gy.copy())
+        if x_data.ndim == 3:
+            x_data[0, :, :] = 0.0
+        else:
+            x_data[:, :] = 0.0
+        with self.assertRaises(ValueError):
+            gradient_check.check_backward(self.det, x_data, y_grad)
+
+    def test_answer_cpu(self):
```

Comment:
How about making x randomized and calculating y with the formula det(A) = ad - bc? This change makes the tests more robust.